compose/config/fields_schema_v2.json

```diff
@@ -41,7 +41,24 @@
   "definitions": {
     "network": {
       "id": "#/definitions/network",
-      "type": "object"
+      "type": "object",
+      "properties": {
+        "driver": {"type": "string"},
+        "driver_opts": {
+          "type": "object",
+          "patternProperties": {
+            "^.+$": {"type": ["string", "number"]}
+          }
+        },
+        "external": {
+          "type": ["boolean", "object"],
+          "properties": {
+            "name": {"type": "string"}
+          },
+          "additionalProperties": false
+        }
+      },
+      "additionalProperties": false
```

Comment:
I think this looks like a good start, but we also support the `ipam` field (https://github.com/docker/compose/blob/master/docs/compose-file.md#network-configuration-reference)